plugins/native/module/vtkF3DQuakeMDLImporter.cxx

```diff
@@ -0,0 +1,528 @@
+  void CreateMesh(const std::vector<unsigned char>& buffer, int offset, mdl_header_t* header)
+  {
+    // Read texture coordinates
+    struct mdl_texcoord_t
+    {
+      int onseam;
+      int s;
+      int t;
+    };
+    const mdl_texcoord_t* texcoords =
+      reinterpret_cast<const mdl_texcoord_t*>(buffer.data() + offset);
+    offset += 12 * header->numVertices;
+
+    // Read triangles
+    struct mdl_triangle_t
+    {
+      int facesfront; // 0 = backface, 1 = frontface
+      int vertex[3];  // vertex indices
+    };
+    const mdl_triangle_t* triangles =
+      reinterpret_cast<const mdl_triangle_t*>(buffer.data() + offset);
+    offset += 16 * header->numTriangles;
+
+    // Simple vertex and normal
+    struct mdl_vertex_t
+    {
+      unsigned char v[3];
+      unsigned char normalIndex;
+    };
+    // Simple frame
+    struct mdl_simpleframe_t // 24 + nbVertices bytes
+    {
+      mdl_vertex_t bboxmin;      // bounding box min
+      mdl_vertex_t bboxmax;      // bounding box max
+      char name[16];             // frame name
+      mdl_vertex_t verts[1024];  // vertex list of the frame, maximum capacity is 1024
+    };
+    // Struct containing frame type and data
+    struct plugin_frame_pointer
+    {
+      const int* type;                 // 0 = simple frame, !0 = group frames
+      const int* nb;                   // "number", size of *time and *frames arrays
+      const float* time;               // time duration for each frame
+      const mdl_simpleframe_t* frames; // simple frame list
+    };
+    // Read frames
+    std::vector<plugin_frame_pointer> framePtr =
+      std::vector<plugin_frame_pointer>(header->numFrames);
+    std::vector<std::vector<int>> frameOffsets = std::vector<std::vector<int>>();
+    for (int i = 0; i < header->numFrames; i++)
+    {
+      framePtr[i].type = reinterpret_cast<const int*>(buffer.data() + offset);
+      if (*framePtr[i].type == 0)
+      {
+        framePtr[i].nb = nullptr;
+        framePtr[i].time = nullptr;
+        framePtr[i].frames = reinterpret_cast<const mdl_simpleframe_t*>(buffer.data() + 4 + offset);
+        offset += 4 + 24 + 4 * header->numVertices;
+      }
+      else
+      {
+        framePtr[i].nb = reinterpret_cast<const int*>(buffer.data() + 4 + offset);
+        framePtr[i].time = reinterpret_cast<const float*>(
+          reinterpret_cast<const void*>(buffer.data() + 16 + offset));
+        framePtr[i].frames = reinterpret_cast<const mdl_simpleframe_t*>(
+          buffer.data() + 16 + 4 * (*framePtr[i].nb) + offset);
+        frameOffsets.emplace_back(std::vector<int>());
+        for (int j = 0; j < *framePtr[i].nb; j++)
+        {
+          frameOffsets[i].emplace_back(
+            16 + 4 * (*framePtr[i].nb) + j * 4 * (header->numVertices + 6) + offset);
+        }
+        offset += 16 + (*framePtr[i].nb) * 4 + (24 + 4 * header->numVertices) * (*framePtr[i].nb);
+      }
+    }
```

Comment:
many arbitrary numbers everywhere in this method. Better to define constexpr with meaningfull names when possible,